### PR TITLE
Migrate commands to typer

### PR DESCRIPTION
**Pull request by LukaszKondrackiDynatrace**

### CODE REVIEW

**Review by Dysio — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,6 +66,12 @@ Alternatively, you can use the Docker image to replicate the environment without
 
 #### With Docker
 
+* Copy toml file to ensure cacheability. Bash\Powershell:
+
+  ```shell
+  cp pyproject.toml pyproject.toml.mod
```
Comment (allgreed) [inline CONTRIBUTING.md:72]: Well... why? I mean... we're missing the `sed` part. But I don't think I get why would we even want to do this...

File: dtcli/scripts/dt.py [hunk 2]
```diff
@@ -200,54 +203,57 @@ def extension_dev():
...
+def genca(
+    ca_cert: str = typer.Option(
+        const.DEFAULT_CA_CERT,
+        help="CA certificate output path"
+    ),
+    ca_key: str = typer.Option(
+        const.DEFAULT_CA_KEY,
+        help="CA key output path"
+    ),
+    ca_subject: str = typer.Option(
```
Comment (allgreed) [inline dtcli/scripts/dt.py:216]: This is just plain ugly, can we do this with an accumulating argument?

File: dtcli/scripts/dt.py [hunk 3]
```diff
@@ -200,54 +203,57 @@ def extension_dev():
...
+        "/CN=Default Extension CA/O=Some Company/OU=Extension CA",
+        callback=validate_parse_subject,
+        help="Certificate subject. Accepted format is /key0=value0/key1=value1/..."
+    ),
+    ca_passphrase: str = typer.Option(
```
Comment (allgreed) [inline dtcli/scripts/dt.py:221]: See other places that mention the security of passing passphrase this way

**Comment by allgreed** [comment]
Or maybe we can do it in 2 steps

File: CONTRIBUTING.md (hunk 1 above)
Comment (LukaszKondrackiDynatrace) [reply CONTRIBUTING.md:72]: Right now If somebody cloned the project and tried following Contributing with Docker instructions, they would get:

=> ERROR [6/8] COPY pyproject.toml.mod /app/pyproject.toml                                                                                                                                                                        0.0s 
------
 > [6/8] COPY pyproject.toml.mod /app/pyproject.toml:
------
failed to compute cache key: "/pyproject.toml.mod" not found: not found

I think that We might want to adress it in some way in Contributing instructions

File: dtcli/scripts/dt.py [hunk 4]
```diff
@@ -200,54 +203,57 @@ def extension_dev():
...
-@click.option("--force", is_flag=True, help="Overwrites already existing CA key and certificate")
-@click.option(
-    "--days-valid",
-    default=const.DEFAULT_CERT_VALIDITY,
-    show_default=True,
-    type=int,
-    help="Number of days certificate will be valid",
-)
-def genca(**kwargs):
+    force: bool = typer.Option(
```
[the remark this answers is not in the document]
Comment (LukaszKondrackiDynatrace) [reply dtcli/scripts/dt.py:237]: You're right it's better use common force option

File: CONTRIBUTING.md (hunk 1 above)
Comment (allgreed) [reply CONTRIBUTING.md:72]: Ok, that's a valid argument. Can we do the sed part as well to maintain parity?

Comment (allgreed) [reply CONTRIBUTING.md:72]: Good point about PowerShell comaptibility, we can issue a comment hinting at the right replacement for `sed`

Comment (allgreed) [reply CONTRIBUTING.md:72]: After careful consideration: let's not put there it's ok.

**Review by allgreed — APPROVED** [review]

**Pull request merged**